### ISSUE TITLE
Add OpenGL Wrapper Library

### ISSUE THREAD

**Issue by excaliburHisSheath** [excaliburHisSheath/gunship-rs#64]
Implement an OpenGL wrapper library like [glutin](https://github.com/tomaka/glium) or [gfx-rs](https://github.com/gfx-rs/gfx) that simplifies working with OpenGL and makes doing so safer and more performant. This work has been started on the [`render_reform`](https://github.com/excaliburHisSheath/gunship-rs/tree/render-reform) branch.

This specific task is for re-implementing the basic set of features that have been built in polygon:

- [x] Create vertex and index buffers and safely fill them with data.
- [x] Specify vertex attributes on the vertex buffer.
- [x] Draw arrays using a vertex buffer or draw elements with a vertex and index buffer.
- [x] Use a safe wrapper to specify various optional configuration values when drawing.
- [x] Safely create shaders from source strings and link them into program objects.
- [x] Connect shader variables to vertex buffer attributes.
- [x] Set values for shader uniforms.

Implementing texture support is not part of this task because it was not fully implemented previously.

**Comment by excaliburHisSheath** [comment]
This work is currently in progress on the [`render-reform`](https://github.com/excaliburHisSheath/gunship-rs/tree/render-reform) branch.

**Comment by excaliburHisSheath** [comment]
Work is complete, just waiting on some integration work before it can be merged into `master`.
